noir-projects/aztec-nr/aztec/src/context/utility_context.nr

```diff
@@ -14,31 +12,35 @@ pub struct UtilityContext {
 
 impl UtilityContext {
     pub unconstrained fn new() -> Self {
-        // We could call these oracles on the getters instead of at creation, which makes sense given that they might
-        // not even be accessed. However any performance gains are minimal, and we'd rather fail early if a user
-        // incorrectly attempts to create a UtilityContext in an environment in which these oracles are not
-        // available.
-        let block_number = get_block_number();
-        let timestamp = get_timestamp();
-        let contract_address = get_contract_address();
-        let version = get_version();
-        let chain_id = get_chain_id();
-        Self { block_number, timestamp, contract_address, version, chain_id }
+        get_utility_context()
```

Comment:
Why not remove the other oracles?

Reply:
I mean that'd be a bigger change as it'd force us to pass the `UtilityContext` around possibly. But I'm not sure that's a bad thing.